sentence_transformers/cross_encoder/CrossEncoder.py

```diff
@@ -527,6 +527,8 @@ def rank(
                 'score': -5.082967,
                 'text': "The 'Harry Potter' series, which consists of seven fantasy novels written by British author J.K. Rowling, is among the most popular and critically acclaimed books of the modern era."}]
         """
+        if self.config.num_labels != 1:
+            raise ValueError("CrossEncoder.rank() only works for models with num_labels=1.")
```

Comment:
Nice!
What do you think, should we maybe also refer people to `CrossEncoder.predict()`?
Something like:
```suggestion
            raise ValueError(
            	"CrossEncoder.rank() only works for models with num_labels=1. "
            	"Consider using CrossEncoder.predict() with input pairs instead."
            )
```
Otherwise people might just stop using Sentence Transformers altogether when `CrossEncoder.rank` fails.

Reply:
true